Entities/Structures/Trampoline/TrampolineLogic.as

```diff
@@ -41,12 +41,29 @@ void onTick(CBlob@ this)
 
 	Vec2f ray = holder.getAimPos() - this.getPosition();
 	ray.Normalize();
-
+	
 	f32 angle = ray.Angle();
-	angle = angle > 135 || angle < 45? (holder.isFacingLeft()? 135 : 45) : 90;
-	angle -= 90;
+	CControls@ controls = holder.getControls();
 
-	this.setAngleDegrees(-angle);
+	if (point.isKeyPressed(key_action1))
+	{
+		// rotate in 45 degree steps
+		angle = Maths::Floor((angle + 22.5f) / 45) * 45;
+		this.setAngleDegrees(-angle);
+	}
+	else if (point.isKeyPressed(key_action2))
+	{
+		// set angle to what was on previous tick
+		angle = this.get_f32("old angle");
+		this.setAngleDegrees(angle);
+	}
```

Comment:
This should be prioritised over left-clicking, so please move this if statement above the previous if statement.

Reply:
done